### PR TITLE
fix: handle missing solutions correctly

### PR DESCRIPTION
**Pull request by ojeytonwilliams**
Rather than creating an [[]] the parser now creates [] which isEmpty().
This makes the test suite check the next challenge for a solution.

In addition, the logic for testing solutions was fixed.

@ShaunSHamilton @nhcarrigan if run `npm run test:full-output` in the curriculum there aren't *too* many failing tests.  If you've got any ideas how to fix them, please go ahead and push commits.

### CODE REVIEW

**Comment by ShaunSHamilton** [comment]
@nhcarrigan I have done most of what I can.

The issues I have added comments you can search with `@nhcarrigan`, and they all pretty much seem to be steps that ask Campers to do something, but that change does not follow through to the rest of the steps. So, I am unsure if you want to delete the step or add the change to all the steps.

**Comment by nhcarrigan** [comment]
I will take a look ASAP. If they aren't blocking, we can go ahead and land this and I'll make a new PR.

File: curriculum/challenges/english/01-responsive-web-design/learn-css-grid-by-building-a-magazine/step-005.md [hunk 1]
```diff
@@ -56,7 +56,7 @@ assert(document.querySelector('p')?.className === 'hero-subtitle');
 Your `p` element should have the text set to `Our efforts to restructure our curriculum with a more project-based focus`.
 
 ```js
-assert(document.querySelector('p')?.textContent === 'Our efforts to restructure our curriculum with a more project-based focus');
+assert.equal(document.querySelector('p')?.textContent?.trim(), 'Our efforts to restructure our curriculum with a more project-based focus');
```
Comment (ShaunSHamilton) [inline curriculum/challenges/english/01-responsive-web-design/learn-css-grid-by-building-a-magazine/step-005.md:59]:
```suggestion
assert.equal(document.querySelector('p')?.textContent?.trim().replace(/\s{2,}/, ' '), 'Our efforts to restructure our curriculum with a more project-based focus');
```

**Review by ShaunSHamilton — CHANGES_REQUESTED** [review]
This change works, but I am not entirely happy with removing most of the whitespace - too lenient?

**Comment by ShaunSHamilton** [comment]
> If they aren't blocking, we can go ahead and land this and I'll make a new PR.

Well, CI will fail for all new PRs (or rebased PRs), because the tests will fail.

File: curriculum/challenges/english/01-responsive-web-design/learn-css-grid-by-building-a-magazine/step-005.md (hunk 1 above)
Comment (nhcarrigan) [reply curriculum/challenges/english/01-responsive-web-design/learn-css-grid-by-building-a-magazine/step-005.md:59]: I think too lenient is better than too strict, in this case - we're not teaching them how to write, we're teaching CSS here.

File: curriculum/challenges/english/01-responsive-web-design/learn-css-flexbox-by-building-a-photo-gallery/step-017.md [hunk 2]
```diff
@@ -75,6 +75,7 @@ body {
   flex-wrap: wrap;
   justify-content: center;
   align-items: center;
+  padding: 0;
```
[the remark this answers is not in the document]
Comment (nhcarrigan) [reply curriculum/challenges/english/01-responsive-web-design/learn-css-flexbox-by-building-a-photo-gallery/step-017.md]: oops

File: curriculum/challenges/english/01-responsive-web-design/learn-intermediate-css-by-building-a-picasso-painting/step-089.md [hunk 3]
```diff
@@ -381,3 +382,13 @@ body {
 --fcc-editable-region--
 
 ```
+
+# --solutions--
+
+```html
+// TODO: Add solution once steps 029 and 070 is fixed!
```
Comment (ShaunSHamilton) [inline curriculum/challenges/english/01-responsive-web-design/learn-intermediate-css-by-building-a-picasso-painting/step-089.md:389]: Now that this is fixed, are you going to add the solutions? Or, shall I?

Comment (nhcarrigan) [reply curriculum/challenges/english/01-responsive-web-design/learn-intermediate-css-by-building-a-picasso-painting/step-089.md:389]: I can push them up

**Comment by ShaunSHamilton** [comment]
I think this can be closed now.

**Pull request closed without merging**